### PR TITLE
Trainer

### PR DESCRIPTION
**Pull request by mousyball**
* [feat] Time profiling to trainer.fit().
* [feat] Add 'to.(device)' feature.
* [feat] Add data_to_device function
* [feat] Use global seed to make the result deterministic

### CODE REVIEW

File: pytorch_trainer/trainer/base_trainer.py [hunk 1]
```diff
@@ -113,6 +121,7 @@ def _loss_parser(self, output):
         total_loss = 0
         for value in output.values():
             total_loss += value
+        # TODO: why should it assign back to output?
```
Comment (JanLin0817) [inline pytorch_trainer/trainer/base_trainer.py:124]: I forgot to remove this line

**Review by JanLin0817 — COMMENTED** [review]
Nice!
